jupyterhub/tests/test_api.py

```diff
@@ -420,6 +422,41 @@ def test_make_admin(app):
     assert user.admin
 
 
+@mark.user
+@mark.gen_test
+def test_set_auth_state(app):
+    auth_state = {'secret': 'hello'}
+    db = app.db
+    name = 'admin'
+    user = find_user(db, name)
+    assert user is not None
+
+    with mock.patch('jupyterhub.user.encrypt') as mock_encrypt:
+        encrypt_future = tFuture()
+        encrypt_future.set_result(b"encrypted_state")
+        mock_encrypt.return_value = encrypt_future
```

Comment:
This whole mocking setup looks pretty ugly to me. Is there a way to make it nicer? Maybe just mock the fetching of the crypto key from the environment?

Reply:
Do you want to enable auth state in the default MockHub? Then you won't need to mock-in enabled auth state. Otherwise, you could create a decicated fixture in conftest.